src/main/kotlin/dev/pfilaretov42/aoc/aoc2023/day4/Scratchcards.kt

```diff
@@ -5,8 +5,37 @@ import kotlin.math.pow
 
 fun main() {
     val inputFile = File("input/2023/day4/input")
-    partOne(inputFile)
-//    partTwo(inputFile)
+//    partOne(inputFile)
+    partTwo(inputFile)
+}
+
+fun partTwo(inputFile: File) {
+    val lineRegex = ":\\s+".toRegex()
+    val cardRegex = "\\s+\\|\\s+".toRegex()
+    val numbersRegex = "\\s+".toRegex()
+
+    val cardsCountMap = mutableMapOf<Int, Int>()
+
+    inputFile.forEachLine { line ->
+        val cardAndNumbers = line.split(lineRegex)
+
+        val cardNumber = cardAndNumbers[0].split(numbersRegex)[1].toInt()
+        val cardsCount = cardsCountMap.merge(cardNumber, 1) { oldValue, _ -> oldValue + 1 }!!
+
+        val numbers = cardAndNumbers[1].split(cardRegex)
+        val winningNumbers = numbers[0].split(numbersRegex).toSet()
+        val numbersToTest = numbers[1].split(numbersRegex).toSet()
```

Comment:
**Suggestion:** It seems like you are splitting the `cardAndNumbers` list into `winningNumbers` and `numbersToTest` using the same regex. You could use destructuring declarations to make this more concise and readable. [enhancement]
```suggestion
        val (winningNumbers, numbersToTest) = cardAndNumbers[1].split(cardRegex).map { it.split(numbersRegex).toSet() }
```